src/sentry/workflow_engine/handlers/condition/event_frequency_handlers.py

```diff
@@ -0,0 +1,245 @@
+        of group_id to the result for that group.
+
+        If comparison_interval is not None, we're making the second query in a
+        percent comparison condition. For example, if the condition is:
+            - num of issues is {}% higher in 1 hr compared to 5 min ago
+        The second query would be querying for num of events from:
+            -  5 min ago to 1 hr 5 min ago
+        """
+        if comparison_interval:
+            current_time -= comparison_interval
+        start, end = self.get_query_window(end=current_time, duration=duration)
+
+        with self.disable_consistent_snuba_mode(duration):
+            result = self.batch_query(
+                group_ids=group_ids,
+                start=start,
+                end=end,
+                environment_id=environment_id,
+            )
+        return result
+
+
+@condition_handler_registry.register(Condition.EVENT_FREQUENCY)
+class EventFrequencyConditionHandler(BaseEventFrequencyConditionHandler):
+    @staticmethod
+    def evaluate_value(value: int, comparison: Any) -> DataConditionResult:
+        return value == json.loads(comparison)["value"]
+
+    @property
+    def intervals(self) -> dict[str, tuple[str, timedelta]]:
+        return STANDARD_INTERVALS
+
+    def batch_query(
+        self, group_ids: set[int], start: datetime, end: datetime, environment_id: int
+    ) -> dict[int, int]:
+        batch_sums: dict[int, int] = defaultdict(int)
+        groups = Group.objects.filter(id__in=group_ids).values(
+            "id", "type", "project_id", "project__organization_id"
+        )
+        error_issue_ids, generic_issue_ids = self.get_error_and_generic_group_ids(groups)
+        organization_id = self.get_value_from_groups(groups, "project__organization_id")
+
+        if error_issue_ids and organization_id:
+            error_sums = self.get_chunked_result(
+                tsdb_function=tsdb.backend.get_sums,
+                model=get_issue_tsdb_group_model(GroupCategory.ERROR),
+                group_ids=error_issue_ids,
+                organization_id=organization_id,
+                start=start,
+                end=end,
+                environment_id=environment_id,
+                referrer_suffix="batch_alert_event_frequency",
+            )
+            batch_sums.update(error_sums)
+
+        if generic_issue_ids and organization_id:
+            generic_sums = self.get_chunked_result(
+                tsdb_function=tsdb.backend.get_sums,
+                # this isn't necessarily performance, just any non-error category
+                model=get_issue_tsdb_group_model(GroupCategory.PERFORMANCE),
+                group_ids=generic_issue_ids,
+                organization_id=organization_id,
+                start=start,
+                end=end,
+                environment_id=environment_id,
+                referrer_suffix="batch_alert_event_frequency",
+            )
+            batch_sums.update(generic_sums)
```

Comment:
@saponifi3d this chunk should be able to handle non-error issues

Reply:
is the only difference the model? thoughts on DRYing up this if / else?

Reply:
yeah can do